usrp3/tools/scripts/viv_sim_project.tcl

```diff
@@ -60,12 +60,20 @@ foreach src_file $design_srcs {
     } elseif [expr [lsearch {.ngc .edif} $src_ext] >= 0] {
         puts "BUILDER: Adding Netlist : $src_file"
         read_edif $src_file
+    } elseif [expr [lsearch {.dat} $src_ext] >= 0] {
+        puts "BUILDER: Adding Data File : $src_file"
+        add_files -fileset $sim_fileset -norecurse $src_file
     } elseif [expr [lsearch {.bd} $src_ext] >= 0] {
-            puts "BUILDER: Adding Block Diagram: $src_file"
-            add_files -norecurse $src_file
+        puts "BUILDER: Adding Block Diagram: $src_file"
+        add_files -norecurse $src_file
     } elseif [expr [lsearch {.bxml} $src_ext] >= 0] {
-            puts "BUILDER: Adding Block Diagram XML: $src_file"
-            add_files -norecurse $src_file
+        puts "BUILDER: Adding Block Diagram XML: $src_file"
+        add_files -norecurse $src_file
+    } elseif [expr [string equal $src_ext ""] == 1] {
+        puts "BUILDER: Adding Folder Glob : $src_file"
+        set glob_all [glob $src_file/*.*]
+        puts "BUILDER: Globbed Files : $glob_all"
+        add_files $glob_all
```

Comment:
Can you check if it is a directory first rather than assuming it's a directory based on the extension? Can `file isdirectory` be used to check this?

Reply:
Good point, probably can do something like that. Sounds like a good idea